conans/client/build/autotools_environment.py

```diff
@@ -144,39 +144,58 @@ def configure(self, configure_dir=None, args=None, build=None, host=None, target
+                         "datarootdir"]
+            help_output = self._help_output(configure_dir)
+            not_available = [flag for flag in all_flags if "--%s" % flag not in help_output]
+
             if use_default_install_dirs:
                 for varname in ["bindir", "sbin", "libexec"]:
-                    if not self._is_flag_in_args(varname, args):
+                    if self._valid_configure_flag(varname, args, not_available):
                         args.append("--%s=${prefix}/%s" % (varname, DEFAULT_BIN))
-                if not self._is_flag_in_args("libdir", args):
+                if self._valid_configure_flag("libdir", args, not_available):
                     args.append("--libdir=${prefix}/%s" % DEFAULT_LIB)
                 for varname in ["includedir", "oldincludedir"]:
-                    if not self._is_flag_in_args(varname, args):
+                    if self._valid_configure_flag(varname, args, not_available):
                         args.append("--%s=${prefix}/%s" % (varname, DEFAULT_INCLUDE))
-                if not self._is_flag_in_args("datarootdir", args):
+                if self._valid_configure_flag("datarootdir", args, not_available):
                     args.append("--datarootdir=${prefix}/%s" % DEFAULT_RES)
 
         with environment_append(pkg_env):
             with environment_append(vars or self.vars):
-                configure_dir = self._adjust_path(configure_dir)
-                command = '%s/configure %s %s' % (configure_dir,
-                                                  args_to_string(args), " ".join(triplet_args))
+                command = '%s/configure %s %s' % (configure_dir, args_to_string(args),
+                                                  " ".join(triplet_args))
                 self._conanfile.output.info("Calling:\n > %s" % command)
-                self._conanfile.run(command,
-                                    win_bash=self._win_bash,
-                                    subsystem=self.subsystem)
+                self._conanfile.run(command, win_bash=self._win_bash, subsystem=self.subsystem)
+
+    def _help_output(self, configure_path):
+        from six import StringIO  # Python 2 and 3 compatible
+        mybuf = StringIO()
+        try:
+            self._conanfile.run("%s/configure --help" % configure_path, output=mybuf)
+        except ConanException as e:
+            self._conanfile.output.warn("Error running `configure --help`: %s" % e)
+            return ""
+        return mybuf.getvalue()
 
     def _adjust_path(self, path):
         if self._win_bash:
             path = unix_path(path, path_flavor=self.subsystem)
         return '"%s"' % path if " " in path else path
 
+    @staticmethod
+    def _valid_configure_flag(varname, args, not_available_flags):
+        return not AutoToolsBuildEnvironment._is_flag_in_args(varname, args) and \
+               varname not in not_available_flags
```

Comment:
Double negation. I think a field ``available_flags`` would be easier to read and understand.